### ISSUE TITLE
TypeError: argument of type 'NoneType' is not iterable

### ISSUE THREAD

**Issue by Cipcu** [dask/dask-kubernetes#165]
Hey,

I run a kubernetes cluster on GCP and wanted to experiment with dask. I deployed a jupyter server with modified image so that it has dask-kubernetes. In the notebook I am able to import dask, dask_kubernetes and so on, however when I execute:

`cluster = KubeCluster.from_yaml('worker-spec.yml')`

I get following  Error:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-4-48f1acf625e7> in <module>
      1 from dask_kubernetes import KubeCluster
      2 
----> 3 cluster = KubeCluster.from_yaml('worker-spec.yml')

/opt/conda/lib/python3.6/site-packages/dask_kubernetes/core.py in from_yaml(cls, yaml_path, **kwargs)
    308             d = yaml.safe_load(f)
    309             d = dask.config.expand_environment_variables(d)
--> 310             return cls.from_dict(d, **kwargs)
    311 
    312     @property

/opt/conda/lib/python3.6/site-packages/dask_kubernetes/core.py in from_dict(cls, pod_spec, **kwargs)
    270         KubeCluster.from_yaml
    271         """
--> 272         return cls(make_pod_from_dict(pod_spec), **kwargs)
    273 
    274     @classmethod

/opt/conda/lib/python3.6/site-packages/dask_kubernetes/core.py in __init__(self, pod_template, name, namespace, n_workers, host, port, env, auth, **kwargs)
    192 
    193         pod_template = clean_pod_template(pod_template)
--> 194         ClusterAuth.load_first(auth)
    195 
    196         self.core_api = kubernetes.client.CoreV1Api()

/opt/conda/lib/python3.6/site-packages/dask_kubernetes/auth.py in load_first(auth)
     68         for auth_instance in auth:
     69             try:
---> 70                 auth_instance.load()
     71             except kubernetes.config.ConfigException as exc:
     72                 logger.debug(

/opt/conda/lib/python3.6/site-packages/dask_kubernetes/auth.py in load(self)
    119     def load(self):
    120         kubernetes.config.load_kube_config(
--> 121             self.config_file, self.context, None, self.persist_config
    122         )
    123 

/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py in load_kube_config(config_file, context, client_configuration, persist_config)
    643     loader = _get_kube_config_loader_for_yaml_file(
    644         config_file, active_context=context,
--> 645         persist_config=persist_config)
    646 
    647     if client_configuration is None:

/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py in _get_kube_config_loader_for_yaml_file(filename, persist_config, **kwargs)
    611         config_dict=kcfg.config,
    612         config_base_path=None,
--> 613         **kwargs)
    614 
    615 

/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py in __init__(self, config_dict, active_context, get_google_credentials, config_base_path, config_persister)
    151         self._user = None
    152         self._cluster = None
--> 153         self.set_active_context(active_context)
    154         self._config_base_path = config_base_path
    155         self._config_persister = config_persister

/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py in set_active_context(self, context_name)
    171     def set_active_context(self, context_name=None):
    172         if context_name is None:
--> 173             context_name = self._config['current-context']
    174         self._current_context = self._config['contexts'].get_with_name(
    175             context_name)

/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py in __getitem__(self, key)
    493 
    494     def __getitem__(self, key):
--> 495         v = self.safe_get(key)
    496         if not v:
    497             raise ConfigException(

/opt/conda/lib/python3.6/site-packages/kubernetes/config/kube_config.py in safe_get(self, key)
    489     def safe_get(self, key):
    490         if (isinstance(self.value, list) and isinstance(key, int) or
--> 491                 key in self.value):
    492             return self.value[key]
    493 

TypeError: argument of type 'NoneType' is not iterable
```

Now, I know what the error means in general, but in this case I have no clue how to debug it. I deploy the jupyter with Helm, and that's the dockerfile I'm using:
```
FROM jupyter/minimal-notebook:177037d09156

SHELL ["/bin/bash", "-c"]


RUN conda install --yes \
    -c conda-forge \
    python-blosc \
    cytoolz \
    dask==1.2.0 \
    nomkl \
    numpy==1.16.2 \
    pandas==0.24.2 \
    tornado==5.1.1\
    tini \
    && conda clean -tipsy \
    && find /opt/conda/ -type f,l -name '*.a' -delete \
    && find /opt/conda/ -type f,l -name '*.pyc' -delete \
    && find /opt/conda/ -type f,l -name '*.js.map' -delete \
    && find /opt/conda/lib/python*/site-packages/bokeh/server/static -type f,l -name '*.js' -not -name '*.min.js' -delete \
    && rm -rf /opt/conda/pkgs

RUN pip install dask-kubernetes
RUN pip install --upgrade "ipython[all]"
RUN pip install --upgrade jupyter

COPY apps /apps
```

Maybe I can get some help in what direction should I look after solution, thanks
Raf

**Comment by Cipcu** [comment]
Was an issue with RoleBinding in kubernetes. Cheers

**Comment by jacobtomlinson** [comment]
Thanks for raising this @Cipcu. Glad to hear you fixed it. If you wouldn't mind writing a little more about how you resolved it I would appreciate it, as it will help others in the future.

**Comment by Cipcu** [comment]
Hey,
I guess the problem was the fact, that the pod with jupyter was missing rights, hence `cluster = KubeCluster.from_yaml('worker-spec.yml')` returned nothing.  Creating a Role and binding it solved the error. 
In the jupyter yaml I added the line `serviceAccountName: <Role>` and after deploying it with helm I was able to start workers from jupyter. Now however, the workers seem not to be able to connect with the scheduler. I have to dig more into it.
